File: Criptografia/Criptografia.java
import java.util.Scanner;

public class Criptografia{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        in.nextLine();
        for(int i = 0; i < n; ++i){
            StringBuilder texto = new StringBuilder(in.nextLine());
            for(int j = 0; j < texto.length(); ++j){
                if(Character.isLetter(texto.charAt(j))) texto.setCharAt(j, (char)(texto.charAt(j) + 3));
            }
            for(int j = 0; j < texto.length() / 2; ++j){
                char aux = texto.charAt(j);
                texto.setCharAt(j, texto.charAt(texto.length() - 1 - j));
                texto.setCharAt(texto.length() - 1 - j, aux);
            }
            for(int j = texto.length() / 2; j < texto.length(); ++j) texto.setCharAt(j, (char)(texto.charAt(j) - 1));
            System.out.println(texto);
        }
        in.close();
    }
}

File: Criptografia/Criptografia.cs
using System;
using System.Text;

public class Criptografia{
    public static void Main(){
        int n = int.Parse(Console.ReadLine());
        for(int i = 0; i < n; ++i){
            StringBuilder texto = new StringBuilder(Console.ReadLine());
            for(int j = 0; j < texto.Length; ++j){
                if(char.IsLetter(texto[j])) texto[j] = (char)(texto[j] + 3);
            }
            for(int j = 0; j < texto.Length / 2; ++j){
                char aux = texto[j];
                texto[j] = texto[texto.Length - 1 - j];
                texto[texto.Length - 1 - j] = aux;
            }
            for(int j = texto.Length / 2; j < texto.Length; ++j) texto[j] = (char)(texto[j] - 1);
            Console.WriteLine(texto);
        }
    }
}